addons/hr_expense/views/hr_expense_views.xml

```diff
@@ -385,19 +413,21 @@
             <field name="arch" type="xml">
                 <form string="Expense Reports" class="o_expense_sheet">
                  <header>
-                    <button name="approve_expense_sheets" states="submit" string="Approve" type="object" groups="hr_expense.group_hr_expense_user" class="oe_highlight o_expense_sheet_approve"/>
+                    <field name="is_user_the_expense_responsible" invisible="1"/>
+                    <button name="approve_expense_sheets" string="Approve" type="object" attrs="{'invisible': ['|', ('state', '!=', 'submit'), ('is_user_the_expense_responsible', '=',False)]}" class="oe_highlight o_expense_sheet_approve"/>
                     <button name="action_sheet_move_create" states="approve" string="Post Journal Entries" type="object" groups="account.group_account_manager" class="oe_highlight o_expense_sheet_post"/>
                     <button name="%(hr_expense.hr_expense_sheet_register_payment_wizard_action)d" type="action" string="Register Payment" class="oe_highlight o_expense_sheet_pay" attrs="{'invisible': [('state', '!=', 'post')]}" context="{'default_amount': total_amount, 'partner_id': address_id}" groups="account.group_account_manager"/>
                     <button name="reset_expense_sheets" states="cancel" string="Resubmit" type="object"/>
-                    <button name="%(hr_expense.hr_expense_refuse_wizard_action)d" states="submit,approve" context="{'hr_expense_refuse_model':'hr.expense.sheet'}" string="Refuse" type="action" groups="hr_expense.group_hr_expense_user" />
+                    <button name="%(hr_expense.hr_expense_refuse_wizard_action)d" context="{'hr_expense_refuse_model':'hr.expense.sheet'}" string="Refuse" type="action" attrs="{'invisible': ['|', ('state', 'not in', ('submit','approve')), ('is_user_the_expense_responsible', '=',False)]}" />
```

Comment:
Generally we avoid hiding buttons for things like "I am not the responsible". It should be based on access rights as on all main applications. What do you think about that change ?